### ISSUE TITLE
language switching - landing page, event pages, login forms

### ISSUE THREAD

**Issue by lucyWMDE** [codingdavinci/relaunch2018#233]
expanded from part 2 of issue #230 

class="language-link is-active" is only applied (to "DE" option) on http://codingdavinci.snater.com/de/ or http://codingdavinci.snater.com/. 
![image](https://user-images.githubusercontent.com/42772674/87548205-74888500-c6ac-11ea-93cb-54105ca2b05d.png)

Once I navigate to the French or English versions of the landing page the language controls no longer work - none of the links are active and none of them are clickable.
![image](https://user-images.githubusercontent.com/42772674/87548162-63d80f00-c6ac-11ea-9771-f2688b37614d.png)

The same seems to be true for single events e.g. http://codingdavinci.snater.com/de/events/saar-lor-lux-2020 German version is fine ("DE" option is active) but once I navigate to the french version of the page, language controls are no longer active or clickable.

The same is ALSO true for 
- single datasets e.g. http://codingdavinci.snater.com/de/daten/atlas-der-grenzen-zwischen-frankreich-und-preussen-1833 
- single project descriptions e.g. http://codingdavinci.snater.com/de/projekte/sounds-without-meaning

German version is fine ("DE" option is active) but once I navigate to the french/english version of the page, language controls are no longer active or clickable.
 

Language switching for Einfacher Seiten/FAQ/Presse/Events overview seems fine, for projekt overview, daten overview, news overview see issue #232

**Comment by lucyWMDE** [comment]
This seems to be the same language switching issue mentioned in issue #216 for
http://codingdavinci.snater.com/de/user/login
http://codingdavinci.snater.com/de/user/password
http://codingdavinci.snater.com/de/user/register 
http://codingdavinci.snater.com/de/user/register?role=institution
http://codingdavinci.snater.com/de/user/reset 

I've cleared cache and tested this multiple times. It's definitely a problem.

**Comment by Snater** [comment]
Closing with #231 and #232. All these indeed seem to be related. I was able to reproduce some strange behaviour, but only on staging (not locally) and only when not logged in.
All this seems to be related to caching—with caching in that context not referring to the browser, but to the Drupal cache. After clearing the cache (`/admin/config/development/performance`) things seemed to work for the moment. That is, why I was also not able to reproduce the issue the last time because I had cleared the cache before trying to reproduce.

I have found a Drupal core issue ticket that seems to describe what we are experiencing here and I applied the alternative method to render the language switcher as described in the issue: https://www.drupal.org/project/drupal/issues/3146564#comment-13741784

Hopefully, this solves our issues with the language switcher.
